### PR TITLE
Pull in more RuneLite changes & abstract away from GLSL a little

### PR DESCRIPTION
**Pull request by ScoreUnder**
Still to do:

* [ ] Add a CPU fallback priority rendering thing
* [x] Add a null priority renderer (just dump vertices indiscriminately into the vbo)
* [ ] Bonus: Try to fix alpha jank
* [x] Bonus: Try to fix large region previews on whatever OS/GPU combo is causing it
* [x] Bonus: Mac compatibility?

### CODE REVIEW

**Comment by ScoreUnder** [comment]
This came out a lot bigger than I expected. But mac compat and finally getting rid of jogamp's annoying focus issues is so worth it.

File: src/main/kotlin/controllers/worldRenderer/Renderer.kt [hunk 1]
```diff
@@ -197,154 +194,149 @@ class Renderer constructor(
...
         val clientCycle = ((System.currentTimeMillis() - clientStart) / 20).toInt() // 50 fps
 
+        priorityRenderer.produceVertices(camera, clientCycle)
+
+        if (textureArrayId == -1) {
+            // lazy init textures as they may not be loaded at plugin start.
+            // this will return -1 and retry if not all textures are loaded yet, too.
+            textureArrayId = textureManager.initTextureArray()
+        }
+//        val textures: Array<TextureDefinition> = textureProvider.getTextureDefinitions()
```
Comment (ConnorDY) [inline src/main/kotlin/controllers/worldRenderer/Renderer.kt:316]: Is this caused by the linting?

**Review by ConnorDY — APPROVED** [review]

File: run [hunk 2]
```diff
@@ -1,2 +1,29 @@
...
+        debug_params="$debug_params -javaagent:lwjglx-debug-1.0.0.jar -Dorg.lwjglx.NO_THROW"
+    fi
+    debug_params="$debug_params -Djoml.debug=true"
+    if [ -e log-debug.xml ]; then
+        debug_params="$debug_params -Dlogback.configurationFile=log-debug.xml"
+    fi
+    shift
+fi
+
+exec "$java" $debug_params --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.desktop/sun.awt=ALL-UNNAMED -jar ./build/libs/osrs-environment-exporter-fat-2.1.0.jar "$@"
```
[the remark this answers is not in the document]
Comment (ScoreUnder) [reply run:29]: Not really but I was tired of having my debug changes locally lol

**Review by ConnorDY — APPROVED** [review]

**Review by ConnorDY — APPROVED** [review]

File: src/main/kotlin/cache/definitions/converters/ObjectToModelConverter.kt [hunk 3]
```diff
@@ -88,15 +89,15 @@ class ObjectToModelConverter(
...
+            modelDefinition = getAndRotateModel(modelIdx, isRotated, type, orientation, modelIds!!)
         }
         if (modelDefinition == null) {
             return null
         }
         return modelDefinition
     }
 
-    private fun ObjectDefinition.getAndRotateModel(modelIdx: Int, isRotated: Boolean, orientation: Int, modelIds: IntArray): ModelDefinition? {
+    private fun ObjectDefinition.getAndRotateModel(modelIdx: Int, isRotated: Boolean, type: Int, orientation: Int, modelIds: IntArray): ModelDefinition? {
```
[the remark this answers is not in the document]
Comment (ScoreUnder) [reply src/main/kotlin/cache/definitions/converters/ObjectToModelConverter.kt:100]: This bakes the vertices at an earlier stage so it's safe. I'll comment on #90 the full data flow we get for the normal renderer

File: src/main/kotlin/controllers/worldRenderer/Renderer.kt (hunk 1 above)
Comment (ScoreUnder) [reply src/main/kotlin/controllers/worldRenderer/Renderer.kt:316]: ctrl+slash leaves comment characters at the start (when removing lines) but manual comments tend to be in-line. i'm not too fussed about leaving it this way but if it's a bother I can change it and/or yeet  the code

**Pull request merged**